src/.clang-tidy

```diff
@@ -0,0 +1,59 @@
+---
```

Comment:
Would there be a github actions to call this?

Reply:
In an ideal world / down the line, we would **enforce** it via a github actions workflow yes. For now, let's not:
* There are a gazillion warnings right now that would need to be eliminated before we do, some of which do not have an automatic fix
* The .clang-tidy config file is not at a point where I feel confident every option enable is something we want

so instead I propose we develop with clang-tidy enabled (you can have your favorite editor including MSVC do code analysis with clang-tidy, I set it up for gViM myself), and take it from there, as in: when touching a file for another purpose, look at the warnings/fixes clang-tidy proposes and maybe fix them or adjust the .clang-tidy config if you feel they aren't good enough.

Once we have a decent feeling that the config is what we want, then we can do a bulk `--fix` (well, using the `run-clang-tidy.py` script on the compile database to be exact) on the entire code base, and see how many warnings we're left dealing with manually.

As a side note, clang-tidy is extensible, meaning you can write your own checks with fixits. You can benefit from the clang AST matchers to do bespoke refactorings (and it's something I have toyed with recently) that aren't as dumb as just using regexes. I made an example for you to look at if you're curious: https://godbolt.org/z/8ffYPrr3z

Reply:
Another note is that cppcheck is definitely still useful: see how https://github.com/openstudiocoalition/OpenStudioApplication/pull/406/checks?check_run_id=3182084911 caught some mistakes I've done here?

Reply:
I finally figured out how to run this in Visual Studio, seems like a good approach to start with